service/task.go

```diff
@@ -418,6 +434,23 @@ func (uis *UIServer) taskPage(w http.ResponseWriter, r *http.Request) {
 	}{uiTask, taskHost, pluginContent, uis.Settings.Jira.Host, permissions, uis.GetCommonViewData(w, r, false, true)}, "base", "task.html", "base_angular.html", "menu.html")
 }
 
+func getAbortedBy(abortedByTaskId string) (*abortedByDisplay, error) {
+	abortedTask, err := task.FindOne(task.ById(abortedByTaskId))
+	if err != nil {
+		return nil, err
+	}
+	buildDisplay, err := build.FindOne(build.ById(abortedTask.BuildId))
+	if err != nil {
+		return nil, err
+	}
+	abortedBy := &abortedByDisplay{
+		TaskDisplayName:     abortedTask.DisplayName,
```

Comment:
My understanding is that if we don't find the task abortedTask will be nil

Reply:
if we don't find the task, then err should equal "error finding task", and it should return return nil, err.

Reply:
but FindOne returns `nil, nil` if the task isn't found
(https://github.com/evergreen-ci/evergreen/blob/a32d4dabaaf91a92e742f30f61f9de7b61e921b4/model/task/db.go#L739)

Reply:
but FindOne returns `nil, nil` if the task isn't found
https://github.com/evergreen-ci/evergreen/blob/a32d4dabaaf91a92e742f30f61f9de7b61e921b4/model/task/db.go#L739

Reply:
thanks! I'll added a check.

Reply:
I think the same will happen if the build isn't found
https://github.com/evergreen-ci/evergreen/blob/defccdb373a08991a744d605ea95c88056baef73/model/build/db.go#L228